### ISSUE TITLE
[❔ other question] Using unbiased gradients with multiple camera views?

### ISSUE THREAD

**Issue by Yingyan-Xu** [mitsuba-renderer/mitsuba2#364]
Is the unbiased gradient supposed to work when solving inverse rendering problems with multiple camera views? I noticed that when I set `unbiased=True`, only the gradients associated with the most recent view (i.e., the last time I call `render(...)` is "remembered".  I'm summing up contributions of all views first and then perform gradient step.

**Comment by Speierers** [comment]
Hi @Yingyan-Xu ,

Indeed this is not possible with the current implementation of `render(..., unbiased=True)`. The AD graph will be wiped out everytime you call this function.

You should be able to work around this by reimplementing it yourself, rendering from every view points within it. Something like that:
```python
with optimizer.disable_gradients():
    for i in range(sensor_count):
            images[i] = _render_helper(scene, spp=spp[0], sensor_index=i)

for i in range(sensor_count):
    image_diffs[i] = _render_helper(scene, spp=spp[1], sensor_index=i)
    ek.reattach(images[i], image_diffs[i])
 ```
 
 If that doesn't work, you could also try to use `render(..., unbiased=True)`as it is, but evaluating/storing the computed gradients after every call to it rather than relying on building a larger AD graph. E.g.
 
 ```python
for i in range(sensor_count):
    img = render(..., unbiased=true)
    loss = compute_loss(img)
    ek.backward(loss)
    grad += ek.grad(my_param) # accumulate gradients rather than building a larger graph
```

**Comment by Yingyan-Xu** [comment]
Thanks a lot!
